src/ulohy/brainfuck.md

```diff
@@ -0,0 +1,49 @@
+# Překladač jazyka Brainfuck
+
+Jazyk [Brainfuck](https://esolangs.org/wiki/Brainfuck) je velmi jednoduchý esoterický programovací
+jazyk obsahující pouze osm operátorů, lineární paměť a adresu aktuální paměťové buňky. Jedna buňka
+odpovídá jednomu bytu.
+
+* Operátor `+` inkrementuje hodnotu aktuální buňky.
+* Operátor `-` dekrementuje hodnotu aktuální buňky.
+* Operátor `>` inkrementuje adresu buňky (po inkrementaci ukazujeme na následující buňku).
+* Operátor `<` dekrementuje adresu buňky (po dekrementaci ukazujeme na předcházející buňku).
+* Operátor `[` uvozuje začátek cyklu. Cyklus probíhá, dokud hodnota buňky adresované v době
+  vyhodnocování podmínky není nulová (adresovanou buňku můžeme měnit uvnitř cyklu).
+* Operátor `]` představuje konec těla cyklu, ale nereprezentuje žádné chování.
+* Operátor `,` přečte jeden byte ze vstupu a uloží hodnotu do aktuálně adresované buňky.
+* Operátor `.` vypíše hodnotu aktuálně adresované buňky jako ASCII znak na standardní výstup.
+
+Všechny ostatní znaky jsou ignorovány.
+
+Naprogramujte interpret jazyka Brainfuck. Interpret (angl. *interpreter*), je překladač, který
+zdrojový kód překládá při každém spuštění cílového programu. Program se tedy nikdy nekompiluje
+do spustitelného binárního souboru.
+
+## Implementace
+
+* Program procházejte znak po znaku a jednotlivé operátory interpretujte.
+* Pokud narazíte na konec cyklu (`]`), stačí se vrátit v textu zpět na odpovídající `[`.
+* Nezapomeňte, že cykly můžou být i vnořené.
+* Paměť lze reprezentovat polem bytů fixní velikosti.
+* Adresu lze reprezentovat indexem nebo ukazatelem.
+* Vstup můžeme číst například pomocí `getc(stdout)`.
+* Výstup můžeme realizovat pomocí funkce `putchar`.
+* Při přístupu mimo alokovanou paměť interpret vypíše chybu a překlad programu skončí.
```

Comment:
It would be nice to add a few simple BF programs from the internet and example input/output pairs.